packages/runtime-plugins/flush-dynamic-remote-plugin/src/plugin.ts

```diff
@@ -0,0 +1,95 @@
+import type { registerRemotes } from '@module-federation/runtime';
+import crypto from 'crypto';
+
+type RegisterRemotesOptions = Parameters<typeof registerRemotes>;
+
+type Remote = RegisterRemotesOptions[0][0];
+type RegisterRemoteOption = RegisterRemotesOptions[1];
+
+const STATUS_CALC_HASH = 1;
+const STATUS_NEED_RELOAD = 0;
+
+type RemoteMap = {
+  [remoteName: string]: {
+    entry: string;
+    lastRegisterTime: number;
+    status?: number;
+    remoteEntryHash?: string;
+  };
+};
+
+const remoteMap: RemoteMap = {};
+const DEFAULT_EXPIRED_TIME = 5 * 60 * 1000;
+
+async function calcRemoteEntryHash(entry: string) {
+  return globalThis
+    .fetch(entry)
+    .then((res) => res.text())
+    .then((entryContent) => {
+      const hash = crypto.createHash('sha256');
```

Comment:
i would use something that is cheap to compute shas. I think before i used md4 or md5 for speed. sha might be quite slow